chevah/chevah-keycert
# Create a self signed certificate with custom attributes.

Scope
=====

This add support to create a self signed certificate with custom attributes.

In previous version the self signed certificate was created with random serial number, but using a fixed set of attributes.


Changes
=======

Add command line argument parser.

Refactored the code to allow for defining key usage and constraints for the CSR generation.

The demo script was updated to have extra options when create a self signed certificate/


How to try and test the changes
===============================

reviewers: @dumol 


1.Self signed cert generation
---------------------------------------


To check the command line options and documentation:

```
$ ./build-keycert/bin/python keycert-demo.py ssl-self-signed --help
```

From the command line I was using he following one liner to generate the cert and then parse it with openssl:


```
$ ./build-keycert/bin/python keycert-demo.py ssl-self-signed --common-name=test --key-usage=critical,key-cert-sign,server-authentication --alternative-name=IP:1.2.3.4,DNS:tol.com --constraints=critical,CA:FALSE | openssl x509 -text

```


2.CSR generation
---------------------------------------


To check the command line options and documentation:

```
$ ./build-keycert/bin/python keycert-demo.py ssl-csr --help
```

To generate a new csr... first delete exiting one, then generate csr and then decode with openssh

```
$ rm server.*
$ ./build-keycert/bin/python keycert-demo.py ssl-csr --common-name bla
$ cat server.csr | openssl req -text
```
